influxdb3_server/src/http.rs

```diff
@@ -193,7 +253,11 @@ where
     async fn write_lp(&self, req: Request<Body>) -> Result<Response<Body>> {
         let query = req.uri().query().ok_or(Error::MissingWriteParams)?;
         let params: WriteParams = serde_urlencoded::from_str(query)?;
-        info!("write_lp to {}", params.db);
+        match validate_db_name(&params.db) {
+            DbName::Valid => info!("write_lp to {}", params.db),
+            DbName::InvalidStartChar => return Err(Error::DbNameInvalidStartChar),
+            DbName::InvalidChar => return Err(Error::DbNameInvalidChar),
+        }
```

Comment:
I think it would be more idiomatic to have `validate_db_name` return `Result`, so that this changes to:
```rust
validate_db_name(&params.db)?;
```
Then, you can separate this `match` logic in to a `From` impl, i.e.,
```rust
impl From<DbName> for Error {
    /* ... */
}
```

Reply:
Since I already had the error I was able to cut out `DbName` entirely and had it be a `Result<()>` and moved the info line to just after the call to `validate_db_name`